pmlc/dialect/stripe/autostencil_pass.cc

```diff
@@ -295,6 +298,38 @@ bool AutoStencil::ValidateStrideOne(BlockArgument* idx, unsigned matrix_idx) {
   return false;
 }
 
+bool AutoStencil::IndexExists(BlockArgument* idx, unsigned tensor_idx) {
+  return tensorIdxs[tensor_idx].find(idx) != tensorIdxs[tensor_idx].end();
+}
+
+// Confirm if idx exists in the right place
+bool AutoStencil::ValidateIndexExistance(BlockArgument* idx, unsigned matrix_idx) {
+  switch (matrix_idx) {
```

Comment:
I think this is a fine approach.  In the old stencil pass, we put all of this into the configuration, but I think that's probably overkill.